src/ipc/ipc4/dai.c

```diff
@@ -240,19 +191,72 @@ void dai_dma_release(struct comp_dev *dev)
 	}
 }
 
+static int dai_init_llp_info(struct comp_dev *dev)
+{
+	struct dai_data *dd = comp_get_drvdata(dev);
+	struct ipc4_copier_module_cfg *copier_cfg;
+	union ipc4_connector_node_id node;
+	uint32_t llp_slot_size;
+	uint32_t max_slot;
+	uint32_t node_id;
+	uint32_t offset;
+	int i;
```

Comment:
We all compile with c99 standard now, no need for such C89 stylings.
Variables should be declared where there are used and in the smallest scope possible, so their lifetime ends and may not be reused elsewhere. Example variable - `node_id` - it should be created in line 232 inside for loop. I know that most compiler implementation won't release stack memory until leaving the function scope but it makes it easier for compiler to optimize. Like line 233 would probably be reduced to:
`if (!mailbox_sw_reg_read(offset))` Please optimize your code where possible.

Reply:
refined, thanks